galsim/compound.py

```diff
@@ -614,13 +727,200 @@ def __init__(self, obj, real_space=None, gsparams=None):
 
         # Save the construction parameters (as they are at this point) as attributes so they
         # can be inspected later if necessary.
-        self.real_space = real_space
-        self.orig_obj = obj
+        self._real_space = real_space
+        self._orig_obj = obj
+        self._gsparams = gsparams
 
-        sbp = galsim._galsim.SBAutoCorrelate(
-                obj.SBProfile, real_space=real_space, gsparams=gsparams)
+        sbp = galsim._galsim.SBAutoCorrelate(obj.SBProfile, real_space, gsparams)
         galsim.GSObject.__init__(self, sbp)
 
         if hasattr(obj,'noise'):
             import warnings
             warnings.warn("Unable to propagate noise in galsim.AutoCorrelation")
+
+    @property
+    def orig_obj(self): return self._orig_obj
+    @property
+    def real_space(self): return self._real_space
+
+    def __repr__(self):
+        return 'galsim.AutoCorrelation(%r, real_space=%r, gsparams=%r)'%(
+                self.orig_obj, self.real_space, self._gsparams)
+
+    def __str__(self):
+        if self.real_space:
+            real = ', real_space=True'
+        else:
+            real = ''
+        return 'galsim.AutoCorrelation(%s%s)'%(self.orig_obj, real)
+
+    def __getstate__(self):
+        d = self.__dict__.copy()
+        del d['SBProfile']
+        return d
+
+    def __setstate__(self, d):
+        self.__dict__ = d
+        self.__init__(self._orig_obj, self._real_space, self._gsparams)
+
+
+
+_galsim.SBAutoCorrelate.__getinitargs__ = lambda self: (
+        self.getObj(), self.isRealSpace(), self.getGSParams())
+_galsim.SBAutoCorrelate.__getstate__ = lambda self: None
+_galsim.SBAutoCorrelate.__setstate__ = lambda self, state: 1
+_galsim.SBAutoCorrelate.__repr__ = lambda self: \
+        'galsim._galsim.SBAutoCorrelate(%r, %r, %r)'%self.__getinitargs__()
+
+
+def Transform(obj, jac=(1.,0.,0.,1.), offset=galsim.PositionD(0.,0.), flux_ratio=1.,
```

Comment:
Why is Transform in compound.py?  The other objects in compound all consist of combinations of two or more GSObjects (admittedly, in some cases these are the same object like in AutoCorrelate).  Deconvolve is  an exception but probably belongs in compound since that's where Convolve lives.

Transform seems like a different kind of beast though.  I think I'd be happier putting it in base.py or even creating transform.py.